src/chapters/classes/constructors.rst

```diff
@@ -0,0 +1,157 @@
+   provide one, the Java compiler will generate an “empty” constructor for
+   you, known as a **default constructor**. For example, when we left out a
+   constructor in our ``HelloWorld`` class above, the compiler created the
+   following constructor for us:
+
+   .. sourcecode:: java
+
+      public HelloWorld() {}
+
+   While this can be convenient, you almost always want to provide a
+   constructor to properly initialize your objects.
+
+Overloading Constructors
+------------------------
+
+We can provide multiple constructors for a given class in order to allow
+for different initialization scenarios. This is known as **constructor
+overloading**. When providing multiple constructors, we must ensure that
+each has a different collection of arguments, as determined by the
+number, order, and type of the constructor arguments.
+
+Let’s expand upon our ``Student`` class.
+
+.. admonition:: Example
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Student {
+
+         private String name;
+         private final int studentId;
+         private int numberOfCredits;
+         private double gpa;
+
+         public Student(String name, int studentId,
+                  int numberOfCredits, double gpa) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = numberOfCredits;
+            this.gpa = gpa;
+         }
+
+         public Student(String name, int studentId) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = 0;
+            this.gpa = 0.0;
+         }
+
+         /* getters and setters omitted */
+
+      }
+
+   The first constructor allows for the creation of ``Student`` objects
+   where the code creating the object provides initial values for each of
+   the fields. The second allows for the creation of ``Student`` objects
+   with only ``name`` and ``studentId``. The first constructor would be
+   most useful for creating a transfer student, where credits and a GPA
+   might already be non-zero. However, for all new students, it would be
+   safe to initialize ``numberOfCredits`` and ``gpa`` to be 0.
+
+   A better way to write the above constructors would be this:
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Student {
+
+         private String name;
+         private final int studentId;
+         private int numberOfCredits;
+         private double gpa;
+
+         public Student(String name, int studentId,
+                  int numberOfCredits, double gpa) {
+            this.name = name;
+            this.studentId = studentId;
+            this.numberOfCredits = numberOfCredits;
+            this.gpa = gpa;
+         }
+
+         public Student(String name, int studentId) {
+            this(name, studentId, 0, 0);
+         }
+
+         /* getters and setters omitted */
+
+      }
+
+   Here, we use ``this()`` to invoke another constructor within the same
```

Comment:
I think this info from the second version of the example warrants its own section header because it clarifies the use of this() as a method. also, it's a bit tricky to decipher which this is being called out, so for that reason, id use line numbers to emphasize the new syntax

Reply:
Done!